ggml/src/ggml-vulkan/vulkan-shaders/mul_mat_vec_q6_k.comp

```diff
@@ -46,11 +46,8 @@ void compute_outputs(const uint32_t first_row, const uint32_t num_rows) {
             const uint ib0 = a_offset / QUANT_K + (first_row+n)*num_blocks_per_row;
             const FLOAT_TYPE d = FLOAT_TYPE(data_a[ib0 + i].d);
 
-            FLOAT_TYPE scales[4];
-            scales[0] = FLOAT_TYPE(data_a[ib0 + i].scales[s_offset + 0]);
-            scales[1] = FLOAT_TYPE(data_a[ib0 + i].scales[s_offset + 2]);
-            scales[2] = FLOAT_TYPE(data_a[ib0 + i].scales[s_offset + 4]);
-            scales[3] = FLOAT_TYPE(data_a[ib0 + i].scales[s_offset + 6]);
+            sccache[ix][itid] = FLOAT_TYPE(data_a[ib0 + i].scales[itid]);
```

Comment:
`ix = tid / 16` is static, and only 0 or 1 on Nvidia/Intel and 0..3 on AMD. So why is the first dimension of `sccache` `BLOCK_SIZE / 16 = 16`? Maybe I don't understand the logic correctly, but something is off here.

Reply:
BLOCK_SIZE is 32 or 64, so that seems ok?

Reply:
You're right, I mixed it up with QK_K.